data-prepper-core/src/main/java/org/opensearch/dataprepper/acknowledgements/DefaultAcknowledgementSet.java

```diff
@@ -118,7 +109,7 @@ public boolean release(final EventHandle eventHandle, final boolean result) {
             if (!pendingAcknowledgments.containsKey(eventHandle) ||
                 pendingAcknowledgments.get(eventHandle).get() == 0) {
                 LOG.warn("Unexpected event handle release");
-                numInvalidReleases.incrementAndGet();
+                metrics.increment(DefaultAcknowledgementSetMetrics.INVALID_RELEASES_METRIC_NAME);
```

Comment:
Should we add the metric to line 125 as well? It appears there is a case where we get to line 125 without emitting a metric for failing to release.

Reply:
No. It is a common case. If a reference is released with returning true/false, it is not an interesting case to keep track of. Only invalid ones are interesting ones.